test/unit-tests/components/views/rooms/UserIdentityWarning-test.tsx

```diff
@@ -0,0 +1,350 @@
+Please see LICENSE files in the repository root for full details.
+*/
+
+import React from "react";
+import { sleep } from "matrix-js-sdk/src/utils";
+import {
+    CryptoEvent,
+    EventType,
+    MatrixClient,
+    MatrixEvent,
+    Room,
+    RoomState,
+    RoomStateEvent,
+    RoomMember,
+} from "matrix-js-sdk/src/matrix";
+import { UserVerificationStatus } from "matrix-js-sdk/src/crypto-api";
+import { act, fireEvent, render, screen, waitFor } from "jest-matrix-react";
+
+import { stubClient } from "../../../../test-utils";
+import { UserIdentityWarning } from "../../../../../src/components/views/rooms/UserIdentityWarning";
+import MatrixClientContext from "../../../../../src/contexts/MatrixClientContext";
+
+const ROOM_ID = "!room:id";
+
+function mockRoom(): Room {
+    const room = {
+        getEncryptionTargetMembers: jest.fn(async () => []),
+        getMember: jest.fn((userId) => {}),
+        roomId: ROOM_ID,
+        shouldEncryptForInvitedMembers: jest.fn(() => true),
+        hasEncryptionStateEvent: jest.fn(() => true),
+    } as unknown as Room;
+
+    return room;
+}
+
+function mockRoomMember(userId: string, name?: string): RoomMember {
+    return {
+        userId,
+        name: name ?? userId,
+        rawDisplayName: name ?? userId,
+        roomId: ROOM_ID,
+        getMxcAvatarUrl: jest.fn(),
+    } as unknown as RoomMember;
+}
+
+function dummyRoomState(): RoomState {
+    return new RoomState(ROOM_ID);
+}
+
+// Get the warning element, given the warning text (excluding the "Learn more" link).
+function getWarningByText(text: string): Element {
+    return screen.getByText((content?: string, element?: Element | null): boolean => {
+        return (
+            !!element &&
+            element.classList.contains("mx_UserIdentityWarning_main") &&
+            element.textContent === text + " Learn more"
+        );
+    });
+}
+
+describe("UserIdentityWarning", () => {
+    let client: MatrixClient;
+    let room: Room;
+
+    beforeEach(async () => {
+        client = stubClient();
+        room = mockRoom();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    // This tests the basic functionality of the component.  If we have a room
+    // member whose identity needs accepting, we should display a warning.  When
+    // the "OK" button gets pressed, it should call `pinCurrentUserIdentity`.
+    it("displays a warning when a user's identity needs approval", async () => {
+        room.getEncryptionTargetMembers = jest.fn(async () => {
+            return [mockRoomMember("@alice:example.org", "Alice")];
+        });
+        const crypto = client.getCrypto()!;
+        crypto["getUserVerificationStatus"] = jest.fn(async () => {
+            return Promise.resolve(new UserVerificationStatus(false, false, false, true));
+        });
+        crypto.pinCurrentUserIdentity = jest.fn();
+        const { container } = render(<UserIdentityWarning room={room} key={ROOM_ID} />, {
```

Comment:
You can declare

```
function renderComponent(room) {
   return render(<UserIdentityWarning room={room} key={ROOM_ID} />, { wrapper: ({ ...rest }) => <MatrixClientContext.Provider value={client} {...rest} /> },
}
```

and use it on your test instead of doing multiple times.